soccer/planning/planner/LineKickPlanner.cpp

```diff
@@ -0,0 +1,204 @@
+            Point targetVel =
+                (aimTarget - targetPos).normalized(*_approachSpeed);
+            targetPos -= targetVel.normalized(Robot_Radius + Ball_Radius * 2);
+
+            _finalApproach = true;
+            RobotInstant goalInstant{Pose{targetPos, 0}, Twist{targetVel, 0},
+                                     RJ::Time{0s}};
+            Trajectory path = CreatePath::simple(startInstant, goalInstant,
+                                                 request.constraints.mot);
+            if (!path.empty()) {
+                path.setDebugText(
+                    "FinalPath" + QString::number(path.duration().count()) +
+                    " " + QString::number(timeToHit.count()) + " " +
+                    QString::number(time.time_since_epoch().count()));
+                PlanAngles(path, startInstant, AngleFns::facePoint(aimTarget),
+                           rotationConstraints);
+                return std::move(path);
+            }
+        }
+
+        if (!prevTrajectory.hit(request.static_obstacles, timeInto) &&
+            timeLeft - 1000ms < timeToHit && _reusePathCount < 10) {
+            _reusePathCount++;
+            Point nearPoint;
+            prevTrajectory.setDebugText("Reuse prevPath");
+            if (ball.estimateTimeTo(prevTrajectory.last().pose.position(),
+                                    &nearPoint) >=
+                RJ::now() + timeLeft - 1000ms) {
+                return reuse(std::move(request));
+            }
+        }
+    }
+
+    std::optional<Trajectory> brutePath = attemptBruteForce(request);
+    if (brutePath) return std::move(*brutePath);
+
+    Point targetVel = (aimTarget - ball.pos).normalized(*_approachSpeed);
+    Point targetPos = ball.pos - targetVel.normalized(Robot_Radius * 3);
+
+    request.dynamic_obstacles.push_back(ball.dynamicObs());
+    RobotInstant targetInstant{Pose{targetPos, 0}, Twist{targetVel, 0},
+                               RJ::Time{0s}};
+    request.motionCommand = PathTargetCommand{targetInstant};
+    Trajectory path = _pathTargetPlanner.plan(std::move(request));
+    path.setDebugText("Gives ups");
+    _targetKickPos = std::nullopt;
+    PlanAngles(path, startInstant, AngleFns::facePoint(aimTarget),
+               rotationConstraints);
+    return std::move(path);
+}
+
+Trajectory LineKickPlanner::planForSlowMovingBall(PlanRequest&& request) {
+    constexpr double ballAvoidDistance = 0.05;
+    Point aimTarget = std::get<LineKickCommand>(request.motionCommand).target;
+    const Ball& ball = request.context->state.ball;
+    Trajectory& prevTrajectory = request.prevTrajectory;
+    RobotInstant startInstant = request.start;
+    RotationConstraints rotationConstraints = request.constraints.rot;
+    Point targetVel = (aimTarget - ball.pos).normalized(*_approachSpeed);
+    Point targetPos;
+    if (std::abs(targetVel.angleBetween(
+            ball.pos - startInstant.pose.position())) > 10 * M_PI / 180.0) {
```

Comment:
So, if our target velocity is more than 10 degrees off from the direction to the goal? Then we should pick a new target point off of that? Honestly a lot of the intuition behind these loses me so more comments would be nice (I know I sound like a broken record at this point).